superset/views/database/views.py

```diff
@@ -117,32 +117,58 @@ def form_post(self, form):
         try:
             utils.ensure_path_exists(config["UPLOAD_FOLDER"])
             csv_file.save(path)
-            table = SqlaTable(table_name=form.name.data)
-            table.database = form.data.get("con")
-            table.database_id = table.database.id
-            table.database.db_engine_spec.create_table_from_csv(form, table)
+            table_name = form.name.data
```

Comment:
This is:
1. Becoming a large try/except block
2. Becoming a large method

Can we break up this logic somehow?

Reply:
I took a second look at this method, and I agree, it's becoming rather sizeable and difficult to maintain. I did consider refactoring this more, but decided against it, as there was already quite a few big changes happening (I already threw in a few integration tests to feel less guilty 
😆). I don't mind taking another stab at cleaning this up, but would prefer to do it in one or even two separate PRs, as I believe it will require a fairly big effort that falls beyond the scope of this fix. However, if there are strong feelings to the contrary I am happy to take another look at refactoring this more.

Reply:
Yes, I'm OK deferring the refactor here. Thanks for the response :)